netty/src/main/java/io/grpc/netty/WriteQueue.java

```diff
@@ -100,6 +100,18 @@ ChannelFuture enqueue(QueuedCommand command, ChannelPromise promise, boolean flu
   }
 
   /**
+   * Enqueue the runnable. It is not safe for another thread to queue an Runnable directly to the
+   * event loop, because it will be out-of-order with writes. This method allows the Runnable to be
+   * processed in-order with writes.
+   */
+  void enqueue(Runnable runnable, boolean flush) {
+    queue.add(new RunnableCommand(runnable));
```

Comment:
Hmm... I'd also need to "support" the setter `QueuedCommand.promise(ChannelPromise)`. It feels about the same to me. I'm going to leave it as-is; this is a pretty straight-forward method and would be easy to change in the future if it became a larger problem.